chat/src/main/java/com/github/twitch4j/chat/events/channel/UserStateEvent.java

```diff
@@ -0,0 +1,131 @@
+package com.github.twitch4j.chat.events.channel;
+
+import com.github.twitch4j.chat.events.AbstractChannelEvent;
+import com.github.twitch4j.common.events.domain.EventChannel;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.Value;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.OptionalInt;
+
+/**
+ * This event gets called when a user joins a channel or sends a PRIVMSG to a channel.
+ */
+@Value
+@Getter
+@EqualsAndHashCode(callSuper = false)
+public class UserStateEvent extends AbstractChannelEvent {
+
+    /**
+     * Hexadecimal RGB color code, or null if it is never set.
+     */
+    private String color;
+
+    /**
+     * The user's display name, or null if it is never set.
+     */
+    private String displayName;
+
+    /**
+     * List of emote sets.
+     */
+    private List<String> emoteSets;
+
+    /**
+     * Badges map of tags to versions.
+     */
+    private Map<String, String> badges;
+
+    /**
+     * Metadata related to the chat badges.
+     */
+    private Map<String, String> badgeInfo;
+
+    /**
+     * Event constructor
+     *
+     * @param channel     The channel that this event originates from.
+     * @param displayName The display version of the user name
+     * @param color       Hexadecimal RGB color code of the user
+     * @param emoteSets   The emote sets the user is entitled to use.
+     * @param badges      Chat badges of the user and the version of each badge
+     * @param badgeInfo   Metadata related to the user chat badges
+     */
+    public UserStateEvent(EventChannel channel, String displayName, String color,
+                          List<String> emoteSets, Map<String, String> badges, Map<String, String> badgeInfo) {
+        super(channel);
+        this.displayName = displayName;
+        this.color = color;
+        this.emoteSets = Collections.unmodifiableList(emoteSets);
+        this.badges = Collections.unmodifiableMap(badges);
+        this.badgeInfo = Collections.unmodifiableMap(badgeInfo);
+    }
+
+    /**
```

Comment:
All getters below can then directly get the information from the IRCMessageEvent - and in case anyone wants to access something else they will be able to directly get that information from the IRCMessageEvent.